### PR TITLE
bgpv1: Add the documentation for using serviceAdvertisements

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Related to https://github.com/cilium/cilium/pull/31245
Related to https://github.com/cilium/cilium/pull/30963
```release-note
Add the documentation for using `serviceAdvertisements`
```

### CODE REVIEW

File: Documentation/network/bgp-control-plane.rst [hunk 1]
```diff
@@ -293,26 +293,127 @@ effect.
 Advertising Service Virtual IPs
 -------------------------------
 
-Type Load Balancer Services
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In Kubernetes, a Service has multiple virtual IP addresses,
+such as ``.spec.clusterIP``, ``.spec.clusterIPs``, ``.status.loadBalancer.ingress[*].ip``
+and ``.spec.externalIPs``.
+The BGP control plane can advertise the virtual IP address of the Service to BGP peers.
+This allows users to directly access th e Service from outside the cluster.
```
Comment (YutaroHayakawa) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
This allows users to directly access the Service from outside the cluster.
```

File: Documentation/network/bgp-control-plane.rst [hunk 2]
```diff
@@ -293,26 +293,127 @@ effect.
...
+The ``.serviceSelector`` field is a label selector that selects Services matching
+the specified ``.matchLabels`` or ``.matchExpressions``.
 
-BGP Control Plane can advertise the ingress IPs
-(``status.loadBalancer.ingress[*].ip``) of a Service of type ``LoadBalancer``
-to the BGP peers. This allows the BGP peers to reach the Service directly from
+
+BGP Control Plane can advertise the virtual IPs of the Services to the BGP peers.
+This allows the BGP peers to reach the Service directly from
 outside the cluster. When your upstream router supports Equal Cost Multi Path
```
Comment (YutaroHayakawa) [inline Documentation/network/bgp-control-plane.rst]: We have a similar phrase above. Maybe we should consolidate?

```
This allows the BGP peers to reach the Service directly from
outside the cluster.
```

File: Documentation/network/bgp-control-plane.rst [hunk 3]
```diff
@@ -293,26 +293,127 @@ effect.
...
+       neighbors:
+       - peerAddress: '10.0.0.1/32'
+         peerASN: 64512
+
+
+Advertising ExternalIP Services
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To access Services outside the cluster, specific BPF parameters need to be enabled,
+see the :ref:`External Access To ClusterIP Services <external_access_to_clusterip_services>` section.
```
Comment (YutaroHayakawa) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
If you wish to use this together with KubeProxyReplacement <reference here>, please make sure the ExternalIP support is enabled.
```

Sorry, actually the `external access to clusterip services` were not required for ExternalIP.

File: Documentation/network/bgp-control-plane.rst [hunk 4]
```diff
@@ -293,26 +293,127 @@ effect.
...
+       neighbors:
+       - peerAddress: '10.0.0.1/32'
+         peerASN: 64512
+
+
+Advertising ClusterIP Services
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To access Services outside the cluster, specific BPF parameters need to be enabled,
+see the :ref:`External Access To ClusterIP Services <external_access_to_clusterip_services>` section.
```
Comment (YutaroHayakawa) [inline Documentation/network/bgp-control-plane.rst]:
```suggestion
If you wish to use this together with KubeProxyReplacement <reference here>, specific BPF parameters need to be enabled. See :ref:`External Access To ClusterIP Services <external_access_to_clusterip_services>` section for how to enable it.
```

**Review by YutaroHayakawa — CHANGES_REQUESTED** [review]
Thanks for working on this! Few comments.

**Review by qmonnet — APPROVED** [review]
Looks good, thank you

**Review by harsimran-pabla — APPROVED** [review]
Thanks for adding the docs 🙏

**Review by YutaroHayakawa — APPROVED** [review]
Thanks!

**Pull request merged**